scripts/o3de/tests/test_repo.py

```diff
@@ -275,3 +275,57 @@ def mocked_open(path, mode, *args, **kwargs):
                     object_set = repo.get_template_json_paths_from_all_cached_repos()
                     assert len(object_set) == 2
         assert True
+
+    def test_validation(self):
+        self.o3de_manifest_data = json.loads(TEST_O3DE_MANIFEST_JSON_PAYLOAD)
+        self.o3de_manifest_data["repos"] = []
+        self.created_files.clear()
+
+        def load_o3de_manifest(manifest_path: pathlib.Path = None) -> dict:
+            return copy.deepcopy(self.o3de_manifest_data)
+
+        def save_o3de_manifest(manifest_data: dict, manifest_path: pathlib.Path = None) -> bool:
+            self.o3de_manifest_data = manifest_data
+            return True
+
+        def mocked_requests_get(url):
+            if isinstance(url, urllib.request.Request):
+                url_str = url.get_full_url()
+            else:
+                url_str = url
+
+            if url_str in ['http://o3de.org/repoA/repo.json']:
+                custom_mock = MagicMock()
+                custom_mock.getcode.return_value = 200
+                custom_mock.read.return_value = 0
+                custom_mock.__enter__.return_value = custom_mock
+            else:
+                raise urllib.error.HTTPError(url_str, 404, "Not found", {}, 0)
+
+            return custom_mock
+
+        def mocked_open(path, mode, *args, **kwargs):
+            file_data = bytes(0)
+            print(f'open {path}')
```

Comment:
probably want to remove this

Reply:
Ah, I knew I'd miss one somewhere eventually :)